src/components/Dashboard/FilingHistoryList.vue

```diff
@@ -1,373 +1,58 @@
 <template>
   <div id="filing-history-list">
     <AddCommentDialog
-      :dialog="addCommentDialog"
-      :filing="currentFiling"
+      :dialog="showAddCommentDialog"
+      :filing="getCurrentFiling"
       @close="hideCommentDialog($event)"
       attach="#filing-history-list"
     />
 
     <DownloadErrorDialog
-      :dialog="downloadErrorDialog"
-      @close="downloadErrorDialog=false"
+      :dialog="showDownloadErrorDialog"
+      @close="mutateDownloadErrorDialog(false)"
       attach="#filing-history-list"
     />
 
     <LoadCorrectionDialog
-      :dialog="loadCorrectionDialog"
-      @exit="loadCorrectionDialog=false"
+      :dialog="showLoadCorrectionDialog"
+      @exit="mutateLoadCorrectionDialog(false)"
       attach="#filing-history-list"
     />
 
     <FileCorrectionDialog
-      :dialog="fileCorrectionDialog"
-      @exit="fileCorrectionDialog=false"
-      @redirect="redirectThisFiling($event)"
+      :dialog="showFileCorrectionDialog"
+      @exit="mutateFileCorrectionDialog(false)"
+      @redirect="redirectFiling($event)"
       attach="#filing-history-list"
     />
 
-    <!-- Alternate Loading Spinner -->
-    <v-fade-transition>
-      <div class="loading-container grayed-out" v-show="isBusy">
-        <div class="loading__content">
-          <v-progress-circular color="primary" size="50" indeterminate />
-          <div class="loading-msg white--text">Fetching data</div>
-        </div>
-      </div>
-    </v-fade-transition>
-
     <!-- Court order notification -->
     <v-card v-if="hasCourtOrders" class="my-6 pa-6" elevation="0">
       <v-icon>mdi-gavel</v-icon>
-      <span class="ml-1">Court order(s) have been filed on this company. Review the filing history for impacts
-        to business information.</span>
+      <span class="ml-1">Court order(s) have been filed on this company. Review the filing history for
+        impacts to business information.</span>
     </v-card>
 
-    <div class="scrollable-container">
-      <v-expansion-panels v-if="showHistoryPanel" v-model="panel">
-        <v-expansion-panel
-          class="align-items-top filing-history-item px-6 py-5"
-          v-for="(filing, index) in historyItems"
+    <div class="scrollable-container" v-if="showHistoryPanel">
+      <v-expansion-panels :value="getPanel">
+        <component
+          v-for="(filing, index) in getFilings"
+          :is="is(filing)"
+          :filing="filing"
+          :index="index"
           :key="index"
-        >
-          <v-expansion-panel-header class="no-dropdown-icon pa-0">
```

Comment:
This expansion panel header is in each filing component.

All the blocks below are also in each filing component, or defaulted (using fallback slots).

Reply:
This is awesome! It is going to be really easy to read/understand the different filings and conditionals for each one.